openassessment/xblock/submission_mixin.py

```diff
@@ -185,14 +192,76 @@ def save_submission(self, data, suffix=''):
         else:
             return {'success': False, 'msg': self._(u"This response was not submitted.")}
 
-    def create_submission(self, student_item_dict, student_sub_data):
+    @XBlock.json_handler
+    def save_files_descriptions(self, data, suffix=''):
+        """
+        Save the descriptions for each uploaded file.
+
+        Args:
+            data (dict): Data should have a single key 'descriptions' that contains
+                the texts for each uploaded file.
+            suffix (str): Not used.
+
+        Returns:
+            dict: Contains a bool 'success' and unicode string 'msg'.
+        """
+        if 'descriptions' in data:
+            descriptions = data['descriptions']
+
+            if isinstance(descriptions, list):
+                all_description_correct = True
+                for description in descriptions:
+                    if not isinstance(description, basestring):
+                        all_description_correct = False
+                        break
+
+                if all_description_correct:
```

Comment:
This could be more succinct and eliminate the `all_description_correct` logic:

`if all(map(lambda description: isinstance(description, basestring), descriptions)):`